model_quantization.py

```diff
@@ -35,7 +35,7 @@ def golomb_bits(v):
 
 def quantize_array(A, q):
     quant_A = (np.abs(A) * q).astype(np.int32)
-    quant_A *= np.sign(A)
+    quant_A *= np.sign(A).astype(np.int32)
```

Comment:
What version of numpy are you using? I wonder if something changed, because this was definitely working. This will allocate an extra temporary array, but should be fine.